xdsl/dialects/experimental/stencil.py

```diff
@@ -517,6 +517,19 @@ class CombineOp(Operation):
     irdl_options = [AttrSizedOperandSegments()]
 
 
+@irdl_op_definition
+class HaloSwapOp(Operation):
+    name = "stencil.halo_swap"
+
```

Comment:
We will want depth here too (we are assuming stencil depth is one, but lots of codes have deeper stencils e.g. 2 or 3 so that's wider halos needed).